app/scripts/modules/core/src/managed/ArtifactsList.tsx

```diff
@@ -134,14 +136,29 @@ function getArtifactStatuses({ environments }: IManagedArtifactVersion): Artifac
   // NOTE: The order in which entries are added to `statuses` is important. The highest priority
   // item must be inserted first.
 
-  const isConstraintPendingManualJudgement = (constraint: IStatefulConstraint) =>
-    constraint.type == 'manual-judgement' && constraint.status == StatefulConstraintStatus.PENDING;
-  const requiresManualApproval = environments.some((environment) =>
-    environment.statefulConstraints?.some(isConstraintPendingManualJudgement),
+  const pendingConstraintTypes = new Set<string>();
+  const failedConstraintTypes = new Set<string>();
+
+  environments.forEach((environment) =>
+    environment.statefulConstraints?.forEach(({ type, status }: IStatefulConstraint) => {
+      if (!isConstraintSupported(type)) {
+        return;
+      }
+
+      if (status === StatefulConstraintStatus.PENDING) {
+        pendingConstraintTypes.add(type);
+      } else if (status === StatefulConstraintStatus.FAIL || status === StatefulConstraintStatus.OVERRIDE_FAIL) {
+        failedConstraintTypes.add(type);
+      }
+    }),
   );
-  if (requiresManualApproval) {
-    statuses.push({ appearance: 'progress', iconName: 'manualJudgement' });
-  }
+
+  pendingConstraintTypes.forEach((type) => {
+    statuses.push({ appearance: 'progress', iconName: getConstraintIcon(type) });
+  });
+  failedConstraintTypes.forEach((type) => {
+    statuses.push({ appearance: 'error', iconName: getConstraintIcon(type) });
+  });
```

Comment:
The way I've setup the logic here is that we will reduce all the constraints across all the environments into a list of their unique types – so let's say there are three pending manual judgements across three environments, we'll show _one single pending manual judgement bubble_. If there were any failed manual judgements, we'd show _one single error manual judgement bubble_. So you'll see that some environments have a constraint in that state, but no context about how many. Same is true for pinning right now.

Longer term we may need to think about supporting quantities on these bubbles so we could e.g. have a pending manual judgement bubble with a little "2" to indicate that there are two environments with that constraint status.